src/ppx/fs.cpp

```diff
@@ -190,4 +191,28 @@ std::filesystem::path GetExternalDataPath()
 }
 #endif
 
+std::filesystem::path GetFullPath(const std::filesystem::path& partialPath, const std::filesystem::path& defaultFolder, const std::filesystem::path& ext, const std::string& regexToReplace, const std::string& replaceString)
+{
+    PPX_ASSERT_MSG(partialPath.string() != "", "Partial path cannot be empty string");
+    PPX_ASSERT_MSG(partialPath.has_filename(), "Partial path cannot be a folder");
+
+    std::filesystem::path fullPath = partialPath;
+
+    if (regexToReplace != "") {
+        std::string fileName    = fullPath.filename().string();
+        std::string newfileName = std::regex_replace(fileName, std::regex(regexToReplace), replaceString);
+        fullPath.replace_filename(std::filesystem::path(newfileName));
+    }
+
+    if (!fullPath.has_parent_path()) {
```

Comment:
Should this be checking for root_path instead?
As-is, it prevents prepending any relative path to the basepath.
For desktop, shouldn't change anything, as prepending CWD doesn't change using the path as-is, but it does impact Android: no way to write to the external storage using a relative path.